### PR TITLE
lint-staged types

### PR DESCRIPTION
**Pull request by saiichihashimoto**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Review by okonet — APPROVED** [review]

File: types/lint-staged/lint-staged-tests.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
+import { type Config } from 'lint-staged';
```
Comment (peterblazejewicz) [inline types/lint-staged/lint-staged-tests.ts:1]:
```suggestion
import type { Config } from 'lint-staged';
```
and TS version can drop

File: types/lint-staged/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+// Type definitions for lint-staged 13.2
+// Project: https://github.com/okonet/lint-staged#readme
+// Definitions by: Andrey Okonetchnikov <https://github.com/okonet>
+//                 Saiichi Hashimoto <https://github.com/saiichihashimoto>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// Minimum TypeScript Version: 4.5
```
Comment (peterblazejewicz) [inline types/lint-staged/index.d.ts:6]:
```suggestion
```
there is nothing specific in definition for TS version, just in tests, see other comment

**Comment by saiichihashimoto** [comment]
Ready for review again!

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@saiichihashimoto thanks!

**Comment by saiichihashimoto** [comment]
Ready to merge

**Pull request merged**